x/stakeibc/module_ibc.go

```diff
@@ -211,7 +186,40 @@ func (im IBCModule) NegotiateAppVersion(
 	return proposedVersion, nil
 }
 
+func (im IBCModule) UnmarshalAck(ctx sdk.Context, acknowledgement []byte) (*channeltypes.Acknowledgement_Result, error) {
+	ack := channeltypes.Acknowledgement_Result{}
+	eventType := "callback"
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			eventType,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+			sdk.NewAttribute(types.AttributeKeyAck, fmt.Sprintf("%v", ack)),
+		),
+	)
+	err := json.Unmarshal(acknowledgement, &ack)
```

Comment:
nit: maybe add a clarifying comment here that the ack can come back as either a success or error type, which is why we first try to unmarshal as a success and then try the error type if that fails